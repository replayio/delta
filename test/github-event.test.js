import { describe, it, expect } from "vitest";

const dotenv = require("dotenv");
const fetch = require("node-fetch");
const fixtures = require("./fixtures/github");
dotenv.config({ path: "./.env.local" });

async function testEvent(action) {
  const payload = fixtures[action];
  if (!payload) {
    console.error("No fixture found for action", action);
    return;
  }
  const eventType = action.split(".")[0];

  let res;

  try {
    res = await fetch(process.env.HOST + "/api/github-event", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "User-Agent": "GitHub-Hookshot/2179efe",
        "X-GitHub-Event": eventType,
        "X-GitHub-Delivery": "f0799a30-8227-11ed-86fa-53c4b61f18df",
        "X-GitHub-Hook-ID": "393511247",
        "X-GitHub-Hook-Installation-Target-ID": "274973",
        "X-GitHub-Hook-Installation-Target-Type": "integration",
      },
      body: JSON.stringify(payload),
    });

    if (res.status > 299) {
      return res.text();
    }

    return res.json();
  } catch (e) {
    console.error("error", e);
  }
}

describe("github event", () => {
  it("workflow completed", async () => {
    const res = await testEvent("workflow_job.completed");
    expect(res.check.output.title).toEqual("0 of 360 snapshots are different");
  });
});
