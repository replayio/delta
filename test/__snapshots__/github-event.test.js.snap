// Vitest Snapshot v1

exports[`github event > format comment w/ subtitle 1`] = `
"**<a href=\\"https://delta.replay.io/project/dcb5df26/?branch=visuals21\\">146 of 360 changed</a>** **(Approved)**
<details>
          <summary>auto-escaped-terminal-expressions.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/fad228792f8e5040d80f536033f766c79a6cab5b32ad357a0066f544b6b124bc.png\\" />
        </details>
<details>
          <summary>go-to-first-line.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/0e44102fbfd1e508522076890d021b3d2414fe5c7110a2be775ab81affc21c48.png\\" />
        </details>
<details>
          <summary>context-menu-position-one.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/03383de872ba6b3d7fd6cebe69bbac8feebf49a2959802358339493415cf5a04.png\\" />
        </details>
<details>
          <summary>filtered-no-results.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/8cca0c265600ed91c2139358da24c5c719734356254678925b37fb24718a0a10.png\\" />
        </details>
<details>
          <summary>line-4-search-result-two-with-breakpoints.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/b29a3365ce805342f69ae2e60432c9449647f1aaa51b751df802b8ede6ba6eb6.png\\" />
        </details>
<details>
          <summary>log-point-analysis-expanded-console.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/c80cffe387c7e69604c4942fe7ce7835af19adad53e70e33baef35fd1b715f06.png\\" />
        </details>
<details>
          <summary>array-getter-after-inspection.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/e55d18a75fe27aa846fbd88e05c0aba9912b82cde0e26db81217dc04e764cc05.png\\" />
        </details>
<details>
          <summary>array-expanded.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/8615aa47f053624fe3cabe813528a39a5794200b6897f41455b72104ec25dcce.png\\" />
        </details>
<details>
          <summary>go-to-last-line.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/a528a1ebc7af93c9ff6c8c143994997ed446820040fba48500effed8695edf1a.png\\" />
        </details>
<details>
          <summary>line-4-column-19-should-log-and-break.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/272f30c0679d37b14f810774609352bb904ac363d21c440fa38453f74208720e.png\\" />
        </details>"
`;

exports[`github event > format comments 1`] = `
"**<a href=\\"https://delta.replay.io/project/dcb5df26/?branch=visuals21\\">146 of 360 changed</a>** 
<details>
          <summary>auto-escaped-terminal-expressions.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/fad228792f8e5040d80f536033f766c79a6cab5b32ad357a0066f544b6b124bc.png\\" />
        </details>
<details>
          <summary>go-to-first-line.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/0e44102fbfd1e508522076890d021b3d2414fe5c7110a2be775ab81affc21c48.png\\" />
        </details>
<details>
          <summary>context-menu-position-one.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/03383de872ba6b3d7fd6cebe69bbac8feebf49a2959802358339493415cf5a04.png\\" />
        </details>
<details>
          <summary>filtered-no-results.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/8cca0c265600ed91c2139358da24c5c719734356254678925b37fb24718a0a10.png\\" />
        </details>
<details>
          <summary>line-4-search-result-two-with-breakpoints.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/b29a3365ce805342f69ae2e60432c9449647f1aaa51b751df802b8ede6ba6eb6.png\\" />
        </details>
<details>
          <summary>log-point-analysis-expanded-console.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/c80cffe387c7e69604c4942fe7ce7835af19adad53e70e33baef35fd1b715f06.png\\" />
        </details>
<details>
          <summary>array-getter-after-inspection.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/e55d18a75fe27aa846fbd88e05c0aba9912b82cde0e26db81217dc04e764cc05.png\\" />
        </details>
<details>
          <summary>array-expanded.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/8615aa47f053624fe3cabe813528a39a5794200b6897f41455b72104ec25dcce.png\\" />
        </details>
<details>
          <summary>go-to-last-line.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/a528a1ebc7af93c9ff6c8c143994997ed446820040fba48500effed8695edf1a.png\\" />
        </details>
<details>
          <summary>line-4-column-19-should-log-and-break.png</summary>
          <img src=\\"https://delta.replay.io/api/snapshot?path=dcb5df26-b418-4fe2-9bdf-5a838e604ec4/272f30c0679d37b14f810774609352bb904ac363d21c440fa38453f74208720e.png\\" />
        </details>"
`;
